core/chains/evm/logpoller/log_poller.go

```diff
@@ -1276,11 +1276,12 @@ func (lp *logPoller) fillRemainingBlocksFromRPC(
 	logPollerBlocks := make(map[uint64]LogPollerBlock)
 	for _, head := range evmBlocks {
 		logPollerBlocks[uint64(head.Number)] = LogPollerBlock{
-			EvmChainId:     head.EVMChainID,
-			BlockHash:      head.Hash,
-			BlockNumber:    head.Number,
-			BlockTimestamp: head.Timestamp,
-			CreatedAt:      head.Timestamp,
+			EvmChainId:           head.EVMChainID,
+			BlockHash:            head.Hash,
+			BlockNumber:          head.Number,
+			BlockTimestamp:       head.Timestamp,
+			FinalizedBlockNumber: head.Number, // always finalized; only matters if this block is returned by LatestBlock()
```

Comment:
Is this safe? `fillRemainingBlocksFromRPC` is called by `GetBlocksRange`, which is public, so in theory it could be called for blocks that are not yet finalized.

Reply:
I see this is marked as resolved, and you've approved, so I assume you figured this out but for the sake of anyone else reviewing:  this will already return an error if any blocks requested are unfinalized, and it's mentioned in the docstring for the method that it should only be used for requesting finalized blocks.